cgi-bin/DW/Controller/Admin/Approve.pm

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/perl
+#
+# DW::Controller::Admin::Approve
+#
+# Interface for screening new accounts for spam content.
+#
+# Authors:
+#      Jen Griffin <kareila@livejournal.com>
+#
+# Copyright (c) 2020 by Dreamwidth Studios, LLC.
+#
+# This program is free software; you may redistribute it and/or modify it under
+# the same terms as Perl itself. For a copy of the license, please reference
+# 'perldoc perlartistic' or 'perldoc perlgpl'.
+#
+
+package DW::Controller::Admin::Approve;
+
+use strict;
+
+use DW::Controller;
+use DW::Routing;
+use DW::Template;
+use DW::FormErrors;
+
+my $privs = [
+    'siteadmin:approvenew',     # just for this page
+    'siteadmin:spamreports',    # include existing antispam team
+    'suspend',                  # include anyone with generic suspend privs
+    'suspend:recent',           # just for this page
+];
+
+DW::Controller::Admin->register_admin_page(
+    '/',
+    path     => 'recent_accounts',
+    ml_scope => '/admin/recent_accounts/review.tt',
+    privs    => $privs
+);
+
+DW::Routing->register_string( '/admin/recent_accounts', \&approve_handler, app => 1 );
+
+sub approve_handler {
+    my ( $ok, $rv ) = controller( form_auth => 1, privcheck => $privs );
+    return $rv unless $ok;
+
+    my $scope = '/admin/recent_accounts/review.tt';
+
+    my $prop = LJ::get_prop( 'user', 'not_approved' );
+    return error_ml( "$scope.error.disabled", { sitenameshort => $LJ::SITENAMESHORT } )
+        unless $prop && LJ::is_enabled('approvenew');
+
+    my $r         = $rv->{r};
+    my $remote    = $rv->{remote};
+    my $form_args = $r->post_args;
+    my $vars      = {};
+
+    $vars->{can_suspend} = 1
+        if $remote->has_priv( 'suspend', '' ) || $remote->has_priv( 'suspend', 'recent' );
+
+    return DW::Template->render_template( 'admin/recent_accounts/review.tt', $vars )
+        unless $r->did_post;
+
+    my $dbr = LJ::get_db_reader();
+    my $sth = $dbr->prepare( "SELECT userid FROM userproplite2 "
```

Comment:
If you are going to use the clustered property table (`userproplite2`), then you need to use cluster readers. Something like,

```perl
fore my $cid ( @LJ::CLUSTERS ) {
    my $dbcr = LJ::get_cluster_reader( $cid ) or die;
    # look for users
}
```

The other option is to not cluster this property, and use the `userproplite` table, but to do that you need to set `cldversion: 0` and `indexed: 0` in the property settings.

Reply:
Hmm, for the way I'm using it, I think it makes sense for it not to be clustered, but I thought it should be indexed if I was going to be searching the column for matches as opposed to fetching single user rows? Can you clarify that for me? The issue of clustering didn't come up in my testing because my dreamhack doesn't use clusters. (I thought the multihomed property designated whether it was clustered or not.)

Reply:
The naming on these properties is definitely wild.

`cldversion` means 'clustered dversion' i.e. if this field is non-zero, the property is clustered and not stored on the global database.

`indexed` has no effect on clustered properties. It applies to global properties (i.e. `cldversion: 0`).

`multihomed` however makes a clustered property _also_ store a copy on the global db. _But_ it also looks crufty and I'm not sure we should ever use it. It used to be used for properties that would be shown on the profile page (such as `icq`) but we stopped setting it and things work, so maybe we delete this feature? heh. 🔪 

So, basically, what you probably want is `cldversion: 0` and `indexed: 1`, that will allow you to do lookups like "where propid = ? and value = 1" and get back everybody who hasn't been approved yet while hitting an index.

FWIW, I think the indexing probably doesn't add much value since the cardinality of values is 2 and the size of the set is O(new users that haven't been reviewed). I would expect that we'd actually spend more time on the index management than we do on filtering out things anyway.